src/ggml.c

```diff
@@ -6431,7 +6422,11 @@ struct ggml_tensor * ggml_mul_mat(
         struct ggml_context * ctx,
         struct ggml_tensor  * a,
         struct ggml_tensor  * b) {
-    GGML_ASSERT(ggml_can_mul_mat(a, b));
+    // hack to admit GEMM custom operator
+    bool mult_mat_conv1d = (a->ne[0] * a->ne[1]) == b->ne[0];
+    bool mult_mat_conv2d = (a->ne[0] * a->ne[1] * a->ne[2]) == b->ne[0];
```

Comment:
You should be able to avoid this hack by using `ggml_reshape` before calling `ggml_mul_mat` in `ggml_conv_xd`